# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/mine/hotfix-clang.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- mine-0.23/cdb-0.75/seek_set.c.vanilla	2023-05-12 19:23:09.910071399 +0200
+++ mine-0.23/cdb-0.75/seek_set.c	2023-05-12 19:23:19.250072073 +0200
@@ -1,4 +1,5 @@
 #include <sys/types.h>
+#include <unistd.h>
 #include "seek.h"
 
 #define SET 0 /* sigh */
--- mine-0.23/cdb-0.75/cdbget.c.vanilla	2023-05-12 19:22:21.255067886 +0200
+++ mine-0.23/cdb-0.75/cdbget.c	2023-05-12 19:22:38.410069124 +0200
@@ -23,7 +23,7 @@
 static struct cdb c;
 char buf[1024];
 
-main(int argc,char **argv)
+int main(int argc,char **argv)
 {
   char *key;
   int r;
--- ./showfile.c.vanilla	2023-05-13 19:48:43.225572382 +0200
+++ ./showfile.c	2023-05-13 19:48:50.140572882 +0200
@@ -20,6 +20,7 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <stdio.h>
+#include <string.h>
 
 #include "mine.h"
 
--- ./libtar-1.2.11/libtar/libtar.c.vanilla	2023-05-13 19:47:59.570569230 +0200
+++ ./libtar-1.2.11/libtar/libtar.c	2023-05-13 19:48:13.325570223 +0200
@@ -13,6 +13,7 @@
 #include <config.h>
 #include <libtar.h>
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
 #include <errno.h>
--- ./libtar-1.2.11/lib/extract.c.vanilla	2023-05-13 19:47:28.362566977 +0200
+++ ./libtar-1.2.11/lib/extract.c	2023-05-13 19:47:35.429567487 +0200
@@ -18,6 +18,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <utime.h>
+#include <string.h>
 
 #ifdef STDC_HEADERS
 # include <stdlib.h>
--- mine-0.23/cdb-0.75/seek_cur.c.vanilla	2023-05-13 19:53:26.109592808 +0200
+++ mine-0.23/cdb-0.75/seek_cur.c	2023-05-13 19:53:39.400593767 +0200
@@ -1,4 +1,6 @@
 #include <sys/types.h>
+#include <unistd.h>
+
 #include "seek.h"
 
 #define CUR 1 /* sigh */
--- mine-0.23/cdb-0.75/cdbdump.c.vanilla	2023-05-13 19:56:38.274606683 +0200
+++ mine-0.23/cdb-0.75/cdbdump.c	2023-05-13 19:56:41.869606943 +0200
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include "uint32.h"
 #include "fmt.h"
 #include "buffer.h"
--- mine-0.23/cdb-0.75/cdbmake.c.vanilla	2023-05-13 19:55:33.595602013 +0200
+++ mine-0.23/cdb-0.75/cdbmake.c	2023-05-13 19:56:17.525605185 +0200
@@ -1,3 +1,5 @@
+#include <unistd.h>
+#include <stdio.h>
 #include "error.h"
 #include "open.h"
 #include "strerr.h"
@@ -36,7 +38,7 @@
 
 static struct cdb_make c;
 
-main(int argc,char **argv)
+int main(int argc,char **argv)
 {
   unsigned int klen;
   unsigned int dlen;
--- mine-0.23/cdb-0.75/cdbdump.c.vanilla	2023-05-13 19:57:44.601611472 +0200
+++ mine-0.23/cdb-0.75/cdbdump.c	2023-05-13 19:57:49.852611851 +0200
@@ -59,7 +59,7 @@
 
 char strnum[FMT_ULONG];
 
-main()
+int main()
 {
   uint32 eod;
   uint32 klen;
--- mine-0.23/cdb-0.75/cdbstats.c.vanilla	2023-05-13 19:58:35.688615161 +0200
+++ mine-0.23/cdb-0.75/cdbstats.c	2023-05-13 19:59:01.491617024 +0200
@@ -1,3 +1,5 @@
+#include <unistd.h>
+#include "str.h"
 #include "uint32.h"
 #include "fmt.h"
 #include "buffer.h"
@@ -68,7 +69,7 @@
 static unsigned long numrecords;
 static unsigned long numd[11];
 
-main()
+int main()
 {
   uint32 eod;
   uint32 klen;
--- mine-0.23/cdb-0.75/cdbtest.c.vanilla	2023-05-13 20:02:17.141631151 +0200
+++ mine-0.23/cdb-0.75/cdbtest.c	2023-05-13 20:02:47.976633377 +0200
@@ -1,3 +1,5 @@
+#include <unistd.h>
+#include "str.h"
 #include "uint32.h"
 #include "fmt.h"
 #include "buffer.h"
@@ -66,7 +67,7 @@
 
 static struct cdb c;
 
-main()
+int main()
 {
   uint32 eod;
   uint32 klen;
--- mine-0.23/cdb-0.75/testzero.c.vanilla	2023-05-13 20:04:37.634641295 +0200
+++ mine-0.23/cdb-0.75/testzero.c	2023-05-13 20:04:57.646642740 +0200
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include "uint32.h"
 #include "scan.h"
 #include "strerr.h"
@@ -14,7 +15,7 @@
 static char data[65536];
 struct cdb_make c;
 
-main(int argc,char **argv)
+int main(int argc,char **argv)
 {
   int fd;
   unsigned long loop;
